code/modules/ai/ai_minds/ai_mind.dm

```diff
@@ -0,0 +1,55 @@
+//A mind that determines things to be done
+//Decisions can be influenced by node weights, surrounding objects in the world and whatever else is happening toit
+//This part handles all the proc calls on things like the parent being provided and atom being walked to
+//This allow gives the ai component signals to register and the proc to be called on the component (which will usually be redirected to the mind)
+
+//Code interaction restrictions: AI mind can only interact with the ai comp whenever the ai comp requests signals
+//It can never interact with the action state as it's a element
+
+/datum/ai_mind
```

Comment:
The name is a confusing mix of two existing concepts (`AI` and `mind`) without being related to neither.
This should be named something like `ai_behavior` while the other datum should be something like `ai_controller`